### PR TITLE
Stage one of moving app settings to app.config

### PR DESCRIPTION
**Pull request by adoublebarrel**
### What is the context of this PR?

[Trello Card](https://trello.com/c/4kjDSdfM/1049-move-app-config-from-settings-py-to-flask-application-factory)

Removed settings from new_relic and jwt_decoder, plus associated
changes for their dependents.

### How to review 
Functionality should be unchanged, imports of settings.py should be reduced, config should be being accessed via the Flask applications config dictionary.

### CODE REVIEW

File: tests/app/authentication/test_jwe.py [hunk 1]
```diff
@@ -10,43 +9,37 @@
...
     def test_valid_jwe(self):
-        decoder = JWTDecryptor()
-        token = decoder.decrypt_jwt_token(VALID_JWE)
+        decoder = JWTDecryptor(*self.decryptor_args)
+        token = decoder.decrypt_jwt_token(VALID_JWE, self.leeway)
         self.assertEqual("jimmy", token.get("user"))
 
     def test_does_not_contain_four_instances_of_full_stop(self):
         jwe = VALID_JWE.replace('.', '', 1)
-        decoder = JWTDecryptor()
```
Comment (collisdigital) [inline tests/app/authentication/test_jwe.py:24]: Nice clean up 👍

File: tests/app/authentication/test_jwe.py [hunk 2]
```diff
@@ -10,43 +9,37 @@
 
 class JWETest(unittest.TestCase):
     def setUp(self):
-        settings.EQ_USER_AUTHENTICATION_SR_PRIVATE_KEY = TEST_DO_NOT_USE_SR_PRIVATE_PEM
-        settings.EQ_USER_AUTHENTICATION_RRM_PUBLIC_KEY = TEST_DO_NOT_USE_RRM_PUBLIC_PEM
-        settings.EQ_USER_AUTHENTICATION_SR_PRIVATE_KEY_PASSWORD = "digitaleq"
+        self.decryptor_args = (
+            TEST_DO_NOT_USE_SR_PRIVATE_PEM,
+            'digitaleq',
```
Comment (collisdigital) [inline tests/app/authentication/test_jwe.py]: Not sure why this was originally declared directly here; might make sense to move it into a 'TEST_DO_NOT_USE_SR_PRIVATE_PEM_PASSWORD'?

File: tests/app/authentication/test_jwt.py [hunk 3]
```diff
@@ -29,206 +28,210 @@
 
 class JWTTest(unittest.TestCase):  # pylint: disable=too-many-public-methods
     def setUp(self):
-        settings.EQ_USER_AUTHENTICATION_SR_PRIVATE_KEY = TEST_DO_NOT_USE_SR_PRIVATE_PEM
-        settings.EQ_USER_AUTHENTICATION_RRM_PUBLIC_KEY = TEST_DO_NOT_USE_PUBLIC_KEY
-        settings.EQ_USER_AUTHENTICATION_SR_PRIVATE_KEY_PASSWORD = "digitaleq"
+        self.decryptor_args = (
+            TEST_DO_NOT_USE_SR_PRIVATE_PEM,
+            'digitaleq',
```
Comment (collisdigital) [inline tests/app/authentication/test_jwt.py]: Same as above

File: tests/app/authentication/test_jwt_decoder.py [hunk 4]
```diff
@@ -10,55 +9,54 @@
 
 class JWTDecodeTest(unittest.TestCase):
     def setUp(self):
-        settings.EQ_USER_AUTHENTICATION_SR_PRIVATE_KEY = TEST_DO_NOT_USE_SR_PRIVATE_PEM
-        settings.EQ_USER_AUTHENTICATION_RRM_PUBLIC_KEY = TEST_DO_NOT_USE_RRM_PUBLIC_PEM
-        settings.EQ_USER_AUTHENTICATION_SR_PRIVATE_KEY_PASSWORD = "digitaleq"
+        self.SR_PRIVATE_KEY = TEST_DO_NOT_USE_SR_PRIVATE_PEM
+        self.RRM_PUBLIC_KEY = TEST_DO_NOT_USE_RRM_PUBLIC_PEM
+        self.SR_PRIVATE_KEY_PASSWORD = 'digitaleq'
```
Comment (collisdigital) [inline tests/app/authentication/test_jwt_decoder.py]: And again

File: tests/app/authentication/test_jwt_decoder_using_rfc_example.py [hunk 5]
```diff
@@ -74,15 +73,19 @@
 # Unit test based of Example A.1 in RFC 7516 (https://tools.ietf.org/html/rfc7516#appendix-A.1)
 class JWTDecoderRFCTest(unittest.TestCase):
     def setUp(self):
-        settings.EQ_USER_AUTHENTICATION_SR_PRIVATE_KEY = private_pem
-        settings.EQ_USER_AUTHENTICATION_RRM_PUBLIC_KEY = public_pem
-        settings.EQ_USER_AUTHENTICATION_SR_PRIVATE_KEY_PASSWORD = "digitaleq"
+        self.decryptor_args = (
+            private_pem,
+            'digitaleq',
```
Comment (collisdigital) [inline tests/app/authentication/test_jwt_decoder_using_rfc_example.py]: Here

File: app/__init__.py [hunk 6]
```diff
@@ -227,4 +230,4 @@ def get_minimized_asset(filename):
 
 
 def restrict_content_length(application):
-    application.config['MAX_CONTENT_LENGTH'] = settings.EQ_MAX_HTTP_POST_CONTENT_LENGTH
+    application.config['MAX_CONTENT_LENGTH'] = application.config['EQ_MAX_HTTP_POST_CONTENT_LENGTH']
```
Comment (jonnyshaw89) [inline app/__init__.py]: Would it make sent to make `EQ_MAX_HTTP_POST_CONTENT_LENGTH` `MAX_CONTENT_LENGTH` so that this doesn't need to be set?

**Review by jonnyshaw89 — CHANGES_REQUESTED** [review]

File: app/__init__.py (hunk 6 above)
Comment (adoublebarrel) [reply app/__init__.py]: That occurred to me too, as it's seconded I'll make the change now

**Review by jonnyshaw89 — APPROVED** [review]

**Comment by jonnyshaw89** [comment]
@adoublebarrel FYI we normally need 2 approvals to merge.

**Comment by collisdigital** [comment]
@adoublebarrel also usually not the author who clicks the merge button but admitted by convention rather than standard; we should update the Dev workflow to explain or agree it isn't important?

**Comment by adoublebarrel** [comment]
The merge button protocol is mentioned in the workflow, this was generally my bad. I was keen to get working on the next set of changes, saw green and didn't properly reference the Dev workflow doc. Lessons learnt and I agree with the workflow as it is.

**Pull request merged**